non-fungible-token/src/lib.rs

```diff
@@ -0,0 +1,224 @@
+	/// The token properties type
+	type TokenData: Parameter + Member + Default + Copy;
+}
+
+decl_error! {
+	/// Error for non-fungible-token module.
+	pub enum Error for Module<T: Trait> {
+		/// No available class ID
+		NoAvailableClassId,
+		/// No available token ID
+		NoAvailableTokenId,
+		/// Token(ClassId, TokenId) not found
+		TokenNotFound,
+		/// Class not found
+		ClassNotFound,
+		/// The operator is not the owner of the token and has no permission
+		NoPermission,
+		/// Arithmetic calculation overflow
+		NumOverflow,
+		/// Can not destroy class
+		/// Total issuance is not 0
+		CannotDestroyClass,
+	}
+}
+
+pub type ClassInfoOf<T> =
+	ClassInfo<<T as Trait>::TokenId, <T as frame_system::Trait>::AccountId, <T as Trait>::ClassData>;
+pub type TokenInfoOf<T> = TokenInfo<<T as frame_system::Trait>::AccountId, <T as Trait>::TokenData>;
+
+decl_storage! {
+	trait Store for Module<T: Trait> as NonFungibleToken {
+		/// Next available class ID.
+		pub NextClassId get(fn next_class_id): T::ClassId;
+		/// Next available token ID.
+		pub NextTokenId get(fn next_token_id): T::TokenId;
+		/// Store class info.
+		///
+		/// Returns `None` if class info not set or removed.
+		pub Classes get(fn classes): map hasher(twox_64_concat) T::ClassId => Option<ClassInfoOf<T>>;
+		/// Store token info.
+		///
+		/// Returns `None` if token info not set or removed.
+		pub Tokens get(fn tokens): double_map hasher(twox_64_concat) T::ClassId, hasher(twox_64_concat) T::TokenId => Option<TokenInfoOf<T>>;
+		/// Token existence check by owner and class ID.
+		pub TokensByOwner get(fn tokens_by_owner): double_map hasher(twox_64_concat) T::AccountId, hasher(twox_64_concat) (T::ClassId, T::TokenId) => Option<()>;
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+	}
+}
+
+impl<T: Trait> Module<T> {
+	/// Create NFT(non fungible token) class
+	pub fn create_class(owner: &T::AccountId, metadata: CID, data: T::ClassData) -> Result<T::ClassId, DispatchError> {
+		let class_id = Self::next_class_id();
+		ensure!(class_id != T::ClassId::max_value(), Error::<T>::NoAvailableClassId);
+
+		let info = ClassInfo {
+			metadata,
+			total_issuance: Default::default(),
+			owner: owner.clone(),
+			data,
+		};
+		Classes::<T>::insert(class_id, info);
+		NextClassId::<T>::mutate(|id| *id += 1.into());
+
+		Ok(class_id)
+	}
+
+	/// Transfer NFT(non fungible token) from `from` account to `to` account
+	pub fn transfer(from: &T::AccountId, to: &T::AccountId, token: (T::ClassId, T::TokenId)) -> DispatchResult {
+		ensure!(Tokens::<T>::contains_key(token.0, token.1), Error::<T>::TokenNotFound);
+		ensure!(
+			TokensByOwner::<T>::contains_key(from.clone(), token),
+			Error::<T>::NoPermission
+		);
+
+		if from == to {
+			return Ok(());
+		}
+
+		Tokens::<T>::mutate(token.0, token.1, |token_info| {
+			if let Some(info) = token_info {
+				info.owner = to.clone();
+			}
+		});
+		TokensByOwner::<T>::remove(from.clone(), token);
```

Comment:
```suggestion
		TokensByOwner::<T>::remove(from, token);
```

the key can be a reference so never need to clone the key. take a reference if needed.